test/provisioner/common.go

```diff
@@ -0,0 +1,62 @@
+package provisioner
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+type patchLabel struct {
+	Op    string            `json:"op"`
+	Path  string            `json:"path"`
+	Value map[string]string `json:"value"`
+}
+
+func AddNodeRoleWorkerLabel(ctx context.Context, clusterName string, cfg *envconf.Config) error {
+	fmt.Printf("Adding worker label to nodes belonging to: %s\n", clusterName)
+
+	config, err := clientcmd.BuildConfigFromFlags("", cfg.KubeconfigFile())
+	if err != nil {
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	nodelist, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	label := map[string]string{"node.kubernetes.io/worker": ""}
+
+	payload := []patchLabel{{
+		Op:    "add",
+		Path:  "/metadata/labels",
+		Value: label,
+	}}
+
+	payloadBytes, _ := json.Marshal(payload)
+	workerStr := clusterName + "-worker"
+
+	for _, node := range nodelist.Items {
+		if strings.Contains(node.Name, workerStr) {
+			_, err = clientset.CoreV1().Nodes().Patch(ctx, node.Name, types.JSONPatchType, payloadBytes, metav1.PatchOptions{})
```

Comment:
If the label already exists, will it fail? If yes then maybe it should check its existence first.

Reply:
afaik, it shouldn't fail, at least with this API. It has not errored out for that reason when I tested it.